src/d/actor/d_a_obj_ohatch.cpp

```diff
@@ -4,72 +4,324 @@
  */
 
 #include "d/actor/d_a_obj_ohatch.h"
+#include "d/res/res_ohatch.h"
 #include "d/d_procname.h"
+#include "d/d_bg_w.h"
 #include "d/d_priority.h"
+#include "d/d_com_inf_game.h"
+#include "f_op/f_op_actor_mng.h"
+#include "m_Do/m_Do_hostIO.h"
+
+namespace {
+static const char l_arcname[] = "Ohatch";
+}
+
+#if VERSION == VERSION_DEMO
+class daObjOhatch_HIO_c : public JORReflexible {
+public:
+    daObjOhatch_HIO_c();
+    virtual ~daObjOhatch_HIO_c() {}
+
+    void genMessage(JORMContext*);
+
+public:
+    /* 0x04 */ s8 m04;
+    /* 0x08 */ f32 m08;
+    /* 0x0C */ f32 m0C;
+    /* 0x10 */ f32 m10;
+    /* 0x14 */ f32 m14;
+    /* 0x18 */ f32 m18;
+    /* 0x1C */ f32 m1C;
+    /* 0x20 */ f32 m20;
+    /* 0x24 */ f32 m24;
+    /* 0x28 */ f32 m28;
+    /* 0x2C */ f32 m2C;
+    /* 0x30 */ f32 m30;
+    /* 0x34 */ f32 m34;
+    /* 0x38 */ u8 m38;
+}; // size = 0x40
+
+static daObjOhatch_HIO_c l_HIO;
+
+daObjOhatch_HIO_c::daObjOhatch_HIO_c() {
+    m04 = -1;
+    m08 = -4096.0f;
+    m0C = -16384.0f;
+    m10 = 0.0f;
+    m14 = -25.0f;
+    m18 = 190.0f;
+    m1C = 1.0f;
+    m20 = 8192.0f;
+    m24 = 131072.0f;
+    m28 = 6.0f;
+    m2C = -64.0f;
+    m30 = 1024.0f;
+    m34 = 8192.0f;
+    m38 = 0;
+}
+#else
+namespace L_HIO {
+static const f32 m04 = -1;
+static const f32 m08 = -4096.0f;
+static const f32 m0C = -16384.0f;
+static const f32 m10 = 0.0f;
+static const f32 m14 = -25.0f;
+static const f32 m18 = 190.0f;
+static const f32 m1C = 1.0f;
+static const f32 m20 = 8192.0f;
+static const f32 m24 = 131072.0f;
+static const f32 m28 = 6.0f;
+static const f32 m2C = -64.0f;
+static const f32 m30 = 1024.0f;
+static const f32 m34 = 8192.0f;
+static const f32 m38 = 0;
```

Comment:
```suggestion
static const u8 m38 = 0;
```
To match the type of the non-const equivalent version